compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -1568,6 +1569,28 @@ class Objects(using Context @constructorOnly):
         throw new Exception("unexpected type: " + tp + ", Trace:\n" + Trace.show)
   }
 
+  /** Widen the escaped value (a method argument or rhs of an assignment)
+   *
+   *  The default widening is 1 for most values, 2 for function values.
+   *  User-specified widening annotations are repected.
+   */
+  def widenEscapedValue(value: Value, expr: Tree): Contextual[Value] =
+    expr.tpe.getAnnotation(defn.InitWidenAnnot) match
```

Comment:
This implements the default as 1 or 2 if there is no annotation, but the default when there is a malformed annotation is always 1.

It would be cleaner to first compute the default widening level (1 or 2), and then do the parsing of the annotation, using that default widening level if annotation parsing fails.

Even cleaner would be a function like:
```scala
def parseAnnotation: Option[Int]
```
that returns `None` if there is no annotation or there is an annotation that cannot be parsed. Then
```scala
parseAnnotation match
  case Some(i) => value.widen(i)
  case None => if ... then value.widen(2) else value.widen(1)
```